### PR TITLE
Dec 23 2021

### PR DESCRIPTION
**Pull request by Rahul4321Kumar**
Custom Registration serializer of dj-rest-auth
1. Done configuration in settings to diable username field
2. Made serializer.py to make our own custom serializer
3. Save data to DB

### CODE REVIEW

File: EMS/ems/urls.py [hunk 1]
```diff
@@ -18,6 +18,6 @@
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('authn/', include('dj_rest_auth.urls')),
-    path('authn/registration/', include('dj_rest_auth.registration.urls'))
+    path('dj-rest-auth/', include('dj_rest_auth.urls')),
+    path('dj-rest-auth/registration/', include('dj_rest_auth.registration.urls'))
```
Comment (ksilawat-deqode) [inline EMS/ems/urls.py]: Change it to, 
```suggestion
    path('auth/', include('dj_rest_auth.urls')),
    path('auth/registration/', include('dj_rest_auth.registration.urls'))
```

File: EMS/users/serializers.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from django.urls import exceptions as url_exceptions
+
+from rest_framework import serializers, exceptions
+
+
+class RegisterSerializer(serializers.Serializer):
+    GENDER_CHOICES = (
+        ("M", "Male"),
+        ("F", "female"),
+    )
```
Comment (ksilawat-deqode) [inline EMS/users/serializers.py]: Make use of this from `models.py`

File: EMS/users/serializers.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            "gender": self.validated_data.get("gender", ""),
+            "password1": self.validated_data.get("password1", ""),
+            "email": self.validated_data.get("email", ""),
+        }
+
+    def save(self, request):
+        adapter = get_adapter()
+        user = adapter.new_user(request)
+        self.cleaned_data = self.get_cleaned_data()
+        print(self.cleaned_data)
```
Comment (ksilawat-deqode) [inline EMS/users/serializers.py]: Don't forget to remove debug logs.

**Review by ksilawat-deqode — CHANGES_REQUESTED** [review]
Change authentication method to use `JWT`

File: EMS/ems/urls.py (hunk 1 above)
Comment (Rahul4321Kumar) [reply EMS/ems/urls.py]: done

File: EMS/users/serializers.py (hunk 2 above)
Comment (Rahul4321Kumar) [reply EMS/users/serializers.py]: done

File: EMS/users/serializers.py (hunk 3 above)
Comment (Rahul4321Kumar) [reply EMS/users/serializers.py]: done

File: EMS/users/models.py [hunk 4]
```diff
@@ -25,18 +25,21 @@ def create_user(self, email, password=None, **extra_fields):
...
 
         if extra_fields.get('is_superuser') is not True:
             raise ValueError('Superuser must have is_superuser=True.')
 
         return self._create_user(email, password, **extra_fields)
 
+GENDER_CHOICES = (
+        ("M", "Male"),
+        ("F", "female"),
+    )
```
Comment (ksilawat-deqode) [inline EMS/users/models.py]: Any constant should be defined inside a separate `constants.py` or if it's just a file-level constant define it at the top after 1 line break from the `imports`

**Review by ksilawat-deqode — CHANGES_REQUESTED** [review]

**Review by ksilawat-deqode — APPROVED** [review]

**Pull request merged**